File: shopping-cart-spring-boot-main/shopping-cart-spring-boot-main/src/main/java/com/ecom/service/CategGeneralService.java
package com.ecom.service;

import com.ecom.model.CategGeneral;
import com.ecom.repository.CategGeneralRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CategGeneralService {

    @Autowired
    private CategGeneralRepository categGeneralRepository;

    // Récupérer toutes les catégories générales
    public List<CategGeneral> getAllCategGenerals() {
        return categGeneralRepository.findAll();
    }

    // Sauvegarder une catégorie générale
    public CategGeneral saveCategGeneral(CategGeneral categGeneral) {
        return categGeneralRepository.save(categGeneral);
    }
}

File: Gestion-Pieces--front-end-main - Copie/src/app/services/CategGeneral.service.ts
// src/app/services/categgeneral.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CategGeneral } from '../models/CategGeneral.model';

@Injectable({
  providedIn: 'root',
})
export class CategGeneralService {
  private apiUrl = 'http://localhost:8081/api/categgeneral'; // URL du backend

  constructor(private http: HttpClient) {}

  // Sauvegarder une catégorie générale
  saveCategGeneral(categGeneral: CategGeneral): Observable<CategGeneral> {
    return this.http.post<CategGeneral>(`${this.apiUrl}/save`, categGeneral);
  }

  // Récupérer toutes les catégories générales
  getAllCategGenerals(): Observable<CategGeneral[]> {
    return this.http.get<CategGeneral[]>(`${this.apiUrl}/all`);
  }
}
